pkg/catalog/catalog.go

```diff
@@ -598,36 +598,25 @@ func (c *Catalog) ListRepositories(ctx context.Context, limit int, prefix, after
 	defer it.Close()
 	// seek for first item
 	afterRepositoryID := graveler.RepositoryID(after)
-	prefixRepositoryID := graveler.RepositoryID(prefix)
-	startPos := prefixRepositoryID
-	if afterRepositoryID > startPos {
-		startPos = afterRepositoryID
-	}
-	if startPos != "" {
-		it.SeekGE(startPos)
```

Comment:
This makes scanning _very_ inefficient.

Reply:
Sorry I fixed it- seek to start position is back in

Reply:
But now we are missing repositories that include the string and are before the prefix (e.g for prefix ple we will never get apple)

Reply:
OK I implemented the substring part. I thought to just call it search string, since that is what the client is trying to do and maybe we will improve the algorithm at some point- let me know what you think